README.md

```diff
@@ -6,103 +6,255 @@ This firmware is in no way associated with The Button Corporation Ltd.
+Opening DFU capable USB device...
+ID 0483:df11
+Run-time device DFU version 011a
+Claiming USB DFU Interface...
+Setting Alternate Setting #0 ...
+Determining device status: state = dfuERROR, status = 10
+dfuERROR, clearing status
+Determining device status: state = dfuIDLE, status = 0
+dfuIDLE, continuing
+DFU mode device DFU version 011a
+Device returned transfer size 2048
+DfuSe interface name: "Internal Flash  "
+dfu-util: Non-valid multiplier ' ', assuming bytes
+Downloading to address = 0x08000000, size = 23288
+Download	[=========================] 100%        23288 bytes
+Download done.
+File downloaded successfully
+```
+
+### Take bttn out of DFU mode
+
+1. Unplug the device from power
+2. Remove the jumper/short of the boot pins
+3. (Optionally) reassemble the unit
+
+### Boot bttn in recovery mode
+
+Plug in the bttn while holding down the button. After 4 seconds it will enter recovery mode and start an open Wi-Fi access point.
+
+1. Next, connect your computer to the `OpenBttn` Wi-Fi network.
+2. Open a browser to [http://192.168.1.1/](http://192.168.1.1/)
+3. Configure the bttn to use your network's Wi-Fi settings.
+
+*You will not need to re-enter recovery mode unless your Wi-Fi network settings have changed.*
+
+### Live configuration of bttn button press URLs
+
+Use a web browser to connect directly to your bttn once it's on your network. You must know your bttn's IP address which you can find out from your router or DHCP server.
+
+Navigate to `http://[bttn_IP_on_your_network]/` and set your short and long-press URLs.
+
+You may reconfigure your bttn as often as desired.
+
+### Built-in socket server
+
+There are several functions available via the socket server. Note that URLs must be encoded to AT format before being passed to the socket server. Refer to [at.js](public/at.js) for more information.
+
+* `dump_config` - return current configuration
+* `blink_leds` - change LED colors in a sequence
+* `url1` - set short-press URL
+* `url2` - set long-press URL
+* `password` - set management password
+
+#### Examples
+
+Set a shell variable for the socket server's URL:
+
+```
+$ export BTTNSOCK="http://[bttn_IP_on_your_network]:8774/socket"
+```
+
+Read configuration for button press actions:
+
+```
+$ curl -w "\n" --data $'auth = PASSWORD\r\ndump_config' $BTTNSOCK
+```
+
+Example output (JSON):
+
+> ```
+> {"url1":"10.1.5.13,/shortPress,4040","url2":"10.1.5.13,/longPress,4040"}
+> ```
+
+Turn on the LEDs (white) for 400ms then then them off:
+
+```
+$ curl -w "\n" --data $'auth = PASSWORD\r\nblink_leds = ffffff;400;0;400' $BTTNSOCK
+```
+
+Example output (plaintext):
+
+> ```
+> Success!
+> ```
 
 ## Current status
 
 ### TODO
 
-* Improve documentation
-
+* Document how LEDs can be specified or individually addressed in `blink_leds` command
+* Document OTA update of Wi-Fi module
+  - Firmware link no longer works
+  - Is this process safe or can it brick the device?
+  - Is it essential? 
+  - What does it improve?
```

Comment:
Here `./resources/bttn-boot-pins-shorting.png` should be `./resources/bttn-boot-pins-shorting.jpg`.